test/sync/src/commonMain/kotlin/io/realm/kotlin/test/mongodb/TestApp.kt

```diff
@@ -28,32 +27,33 @@ import io.realm.kotlin.mongodb.App
 import io.realm.kotlin.mongodb.AppConfiguration
 import io.realm.kotlin.mongodb.Credentials
 import io.realm.kotlin.mongodb.User
-import io.realm.kotlin.test.mongodb.util.AdminApi
-import io.realm.kotlin.test.mongodb.util.AdminApiImpl
-import io.realm.kotlin.test.mongodb.util.defaultClient
+import io.realm.kotlin.test.mongodb.util.AppAdmin
+import io.realm.kotlin.test.mongodb.util.AppAdminImpl
+import io.realm.kotlin.test.mongodb.util.AppServicesClient
+import io.realm.kotlin.test.mongodb.util.BaasApp
+import io.realm.kotlin.test.mongodb.util.Service
+import io.realm.kotlin.test.mongodb.util.TestAppInitializer.initializeDefault
 import io.realm.kotlin.test.platform.PlatformUtils
 import io.realm.kotlin.test.util.TestHelper
 import kotlinx.coroutines.CoroutineDispatcher
 
-const val COMMAND_SERVER_BASE_URL = "http://127.0.0.1:8888"
+const val TEST_APP_PARTITION = "test-app-partition" // With Partion-based Sync
+const val TEST_APP_FLEX = "test-app-flex" // With Flexible Sync
+
 const val TEST_SERVER_BASE_URL = "http://127.0.0.1:9090"
-const val TEST_APP_1 = "testapp1" // With Partion-based Sync
-const val TEST_APP_2 = "testapp2" // Copy of Test App 1
-const val TEST_APP_FLEX = "testapp3" // With Flexible Sync
 
 /**
- * This class merges the classes `App` and `AdminApi` making it easier to create an App that can be
+ * This class merges the classes [App] and [AppAdmin] making it easier to create an App that can be
  * used for testing.
  *
  * @param app gives access to the [App] class delegate for testing purposes
  * @param debug enable trace of command server and rest api calls in the test app.
  */
-class TestApp private constructor(
-    val app: App,
-    dispatcher: CoroutineDispatcher = singleThreadDispatcher("test-app-dispatcher"),
-    debug: Boolean = false
-) : App by app,
-    AdminApi by (runBlocking(dispatcher) { AdminApiImpl(TEST_SERVER_BASE_URL, app.configuration.appId, debug, dispatcher) }) {
+open class TestApp private constructor(
+    pairAdminApp: Pair<App, AppAdmin>
+) : App by pairAdminApp.first, AppAdmin by pairAdminApp.second {
+
+    val app: App = pairAdminApp.first
```

Comment:
Not 100% sure I understand this one as we already have `: App by pairAdminApp.first`, so  `TestApp` is already the app?

Reply:
It is not needed.

Reply:
It is actually needed, it is used by `UserTests::getApp` to validate that the user is using the same app as the test app.